library/src/main/java/com/schibsted/spain/barista/action/BaristaListActions.kt

```diff
@@ -0,0 +1,109 @@
+package com.schibsted.spain.barista.action
+
+import android.support.annotation.IdRes
+import android.support.test.InstrumentationRegistry
+import android.support.test.espresso.AmbiguousViewMatcherException
+import android.support.test.espresso.Espresso.onData
+import android.support.test.espresso.Espresso.onView
+import android.support.test.espresso.FailureHandler
+import android.support.test.espresso.NoMatchingViewException
+import android.support.test.espresso.ViewAction
+import android.support.test.espresso.action.ViewActions.scrollTo
+import android.support.test.espresso.assertion.ViewAssertions.matches
+import android.support.test.espresso.contrib.RecyclerViewActions.actionOnItemAtPosition
+import android.support.test.espresso.contrib.RecyclerViewActions.scrollToPosition
+import android.support.test.espresso.matcher.ViewMatchers.*
+import android.support.v7.widget.RecyclerView
+import android.support.v7.widget.RecyclerView.ViewHolder
+import android.widget.ListView
+import com.schibsted.spain.barista.internal.failurehandler.SpyFailureHandler
+import com.schibsted.spain.barista.internal.failurehandler.withFailureHandler
+import com.schibsted.spain.barista.internal.viewaction.ClickChildAction.clickChildWithId
+import com.schibsted.spain.barista.internal.viewaction.PerformClickAction.clickUsingPerformClick
+import org.hamcrest.Matchers.*
+
+object BaristaListActions {
+
+    @JvmStatic
+    @JvmOverloads
+    fun clickListItem(@IdRes id: Int? = null, position: Int) {
+        performMagicAction(id, position,
+                recyclerAction = actionOnItemAtPosition<ViewHolder>(position, clickUsingPerformClick()),
+                listViewAction = clickUsingPerformClick()
+        )
+    }
+
+    @JvmStatic
+    @JvmOverloads
+    fun scrollListToPosition(@IdRes id: Int? = null, position: Int) {
+        performMagicAction(id, position,
+                recyclerAction = scrollToPosition<ViewHolder>(position),
+                listViewAction = scrollTo()
+        )
+    }
+
+    @JvmStatic
+    @JvmOverloads
+    fun clickListItemChild(@IdRes id: Int? = null, position: Int, @IdRes childId: Int) {
+        performMagicAction(id, position,
+                recyclerAction = actionOnItemAtPosition<ViewHolder>(position, clickChildWithId(childId)),
+                listViewAction = clickChildWithId(childId)
+        )
+    }
+
+    private fun performMagicAction(@IdRes id: Int?, position: Int, recyclerAction: ViewAction, listViewAction: ViewAction) {
+        verifyOneSingleMatch(id)
+
+        val spyFailureHandler = SpyFailureHandler()
+        try {
+            performOnRecycler(id, spyFailureHandler, recyclerAction)
+        } catch (noRecyclerMatching: NoMatchingViewException) {
+            performOnListView(spyFailureHandler, id, position, listViewAction)
```

Comment:
Just a mimimi, parameters order should be the same as `performOnRecycler` method, the first one has to be the `id`